src/MizarBase/include/MizarBase/Time.h

```diff
@@ -11,55 +11,36 @@
 #include "OrbitBase/Typedef.h"
 #include "stdint.h"
 
-namespace orbit_mizar_base {
-
-// wraps `uint64_t` bears semantics of time in nanoseconds and implements a non-wrapping addition
-// TODO(b/242038718) Remove and rely on the explicitly allowed Typedefs arithmetics customisation
-struct NonWrappingNanoseconds {
-  friend NonWrappingNanoseconds operator+(NonWrappingNanoseconds a, NonWrappingNanoseconds b) {
-    const uint64_t sum = a.value + b.value;
-    if (sum < a.value || sum < b.value) return {std::numeric_limits<uint64_t>::max()};
-    return {sum};
-  }
-
-  friend NonWrappingNanoseconds operator-(NonWrappingNanoseconds a, NonWrappingNanoseconds b) {
-    ORBIT_CHECK(a.value >= b.value);
-    return {a.value - b.value};
-  }
+namespace orbit_mizar_base_internal {
 
-  friend NonWrappingNanoseconds operator*(NonWrappingNanoseconds a, uint64_t times) {
-    if (std::numeric_limits<uint64_t>::max() / times < a.value) {
-      return {std::numeric_limits<uint64_t>::max()};
-    }
-    return {a.value * times};
-  }
+static inline uint64_t NonWrappingSum(uint64_t a, uint64_t b) {
```

Comment:
No `static` here as this would make this a hidden symbol. `inline` already makes is a weak symbol (which is what we want)